src/software/ai/hl/stp/action/action.cpp

```diff
@@ -43,6 +43,11 @@ std::unique_ptr<Intent> Action::getNextIntent()
     return next_intent;
 }
 
+std::optional<Robot> Action::getRobot()
+{
+    return robot;
+}
+
 void Action::calculateNextIntentWrapper(IntentCoroutine::push_type &yield)
```

Comment:
Can we not delete this and the IntentCoroutine typedef?

Reply:
Actions still run coroutines, it's just that we get the intents from them in stp now, rather then in tactics. So we still need both.

Reply:
So we're using another layer of coroutines to make tactics return actions?